pkg/input/remoteread/remoteread_test.go

```diff
@@ -0,0 +1,135 @@
+package remoteread
+
+import (
+	"context"
+	"fmt"
+	"github.com/cortexproject/cortex/integration/e2e"
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/pkg/timestamp"
+	"github.com/thanos-community/obslytics/pkg/input"
+	http_util "github.com/thanos-io/thanos/pkg/http"
+	"github.com/thanos-io/thanos/pkg/testutil"
+	"github.com/thanos-io/thanos/test/e2e/e2ethanos"
+	"math"
+	"testing"
+	"time"
+)
+
+// defaultPromConfig returns Prometheus config that sets Prometheus to:
+// * expose 2 external labels, source and replica.
+// * scrape fake target. This will produce up == 0 metric which we can assert on.
+// * optionally remote write endpoint to write into.
+func defaultPromConfig(name string, replica int, remoteWriteEndpoint, ruleFile string) string {
+	config := fmt.Sprintf(`
+global:
+  external_labels:
+    prometheus: %v
+    replica: %v
+scrape_configs:
+- job_name: 'myself'
+  # Quick scrapes for test purposes.
+  scrape_interval: 1s
+  scrape_timeout: 1s
+  static_configs:
+  - targets: ['localhost:9090']
+`, name, replica)
+
+	if remoteWriteEndpoint != "" {
+		config = fmt.Sprintf(`
+%s
+remote_write:
+- url: "%s"
+  # Don't spam receiver on mistake.
+  queue_config:
+    min_backoff: 2s
+    max_backoff: 10s
+`, config, remoteWriteEndpoint)
+	}
+
+	if ruleFile != "" {
+		config = fmt.Sprintf(`
+%s
+rule_files:
+-  "%s"
+`, config, ruleFile)
+	}
+
+	return config
+}
+
+func TestRemoteReadInput_Open(t *testing.T) {
+	t.Parallel()
+
+	s, err := e2e.NewScenario("unit_test_remoteread")
+	testutil.Ok(t, err)
+	t.Cleanup(e2ethanos.CleanScenario(t, s))
+
+	prom, _, err := e2ethanos.NewPrometheus(s.SharedDir(), s.NetworkName(), defaultPromConfig("test", 0, "", ""), e2ethanos.DefaultPrometheusImage())
+	testutil.Ok(t, err)
+	testutil.Ok(t, s.StartAndWaitReady(prom))
+
+	testutil.Ok(t, prom.WaitSumMetricsWithOptions(e2e.Greater(512), []string{"prometheus_tsdb_head_samples_appended_total"}, e2e.WaitMissingMetrics))
+
+	t.Run("test remote read input", func(t *testing.T) {
+		ctx, cancel := context.WithTimeout(context.Background(), 3*time.Minute)
+		defer cancel()
+
+		inConfig := input.InputConfig{
+			Endpoint: "http://" + prom.HTTPEndpoint() + "/api/v1/read",
+			TLSConfig: http_util.TLSConfig{
+				InsecureSkipVerify: true,
+			},
+		}
+		remoteReadInput, err := NewRemoteReadInput(nil, inConfig)
+		testutil.Ok(t, err)
+
+		minT := time.Now().Add(-6 * time.Hour)
+		maxT := timestamp.Time(math.MaxInt64)
+
+		inSeriesParams := input.SeriesParams{
+			Matchers: []*labels.Matcher{
+				labels.MustNewMatcher(labels.MatchEqual, "__name__", "prometheus_tsdb_head_series"),
+			},
+			MinTime: minT,
+			MaxTime: maxT,
+		}
+
+		inSeriesIter, err := remoteReadInput.Open(ctx, inSeriesParams)
+		testutil.Ok(t, err)
+
+		// Go to the first series
```

Comment:
All of those comments have to be fixed for green CI

Reply:
fixed :+1: